src/types.ts

```diff
@@ -67,3 +67,12 @@ export interface EachRepositoryInterface<O> {
 export interface GetInstallationOctokitInterface<O> {
   (installationId: number): Promise<O>;
 }
+
+export interface GetInstallationUrlOptions {
+  state?: string;
+  target_id?: number;
+}
+
+export interface GetInstallationUrlInterface {
+  (options: GetInstallationUrlOptions): Promise<string>;
```

Comment:
I think we should make the argument optional. Once types allow for it, I would update the tests to remove the empty `{}`

```suggestion
  (options?: GetInstallationUrlOptions): Promise<string>;
```

Reply:
Sounds good! I kept it required to match `getWebFlowAuthorizationUrl` ([link](https://github.com/octokit/oauth-methods.js/blob/a9d71a8f53464081d1092de04cdbdc54274105ae/src/get-web-flow-authorization-url.ts#L40-L42)), but not really sure why it's required there either

Reply:
updated to be optional

Reply:
> Sounds good! I kept it required to match `getWebFlowAuthorizationUrl` ([link](https://github.com/octokit/oauth-methods.js/blob/a9d71a8f53464081d1092de04cdbdc54274105ae/src/get-web-flow-authorization-url.ts#L40-L42)), but not really sure why it's required there either

a `clientId` is required because it's part of the URL. This is a stateless function, while here we add a method to which we can pass internal state from the `app` instance.

Reply:
Ah I permalinked the wrong code - the method I'm referencing is [here](https://github.com/octokit/oauth-app.js/blob/50f0b0b7694ff12707dd3c656a15d145ab782be7/src/methods/get-web-flow-authorization-url.ts#L35-L45), and is a wrapper around that `getWebFlowAuthorizationUrl`. In my app, I have to invoke it as `githubApp.oauth.getWebFlowAuthorizationUrl({})`

Reply:
It's been a while, but I think I couldn't figure out how to make `options` optional while supporting the whole OAuth App vs GitHub App type dance ... but I'm not 100% sure. If you want to look into it, I agree `options` should be optional there